core.js

```diff
@@ -111,7 +111,10 @@ module.exports = function (xhr) {
       // With jQuery.ajax's syntax.
       var request = options.xhr = options.xhrImplementation(ajaxSettings, function (err, resp, body) {
           if (err || resp.statusCode >= 400) {
-              if (options.error) return options.error(resp, 'error', (err? err.message : body));
+              if (options.error) {
+                  var message = (err? err.message : (body || "HTTP"+resp.statusCode));
+                  return options.error(resp, 'error', message, resp);
```

Comment:
I'd like to blame it on autocomplete or something, but that's just me getting distracted for a moment ;)